minio/thread_pool.py

```diff
@@ -61,8 +62,7 @@ def __init__(self, num_threads):
         self.results_queue = queue()
         self.exceptions_queue = queue()
         self.tasks_queue = queue(num_threads)
```

Comment:
@nnehdi you need to remove queue size restriction here.. otherwise multipart PUT won't work. The reason is that the  code in this PR will fill tasks_queue with all tasks before starting to execute them.

Reply:
I've tested the existing code (without @vadmeste's suggested correction above) and verified that multi-part uploads fails when more than `num_threads` parts are involved in the multi-part upload. That is, when the `tasks-queue` size is limited with `num_threads`, 3 in our case, the multi-part upload hangs with files bigger than 15MB (3 x 5MB - multi-part size limit) since `tasks_queue` gets full after 3rd part is `tasks_queue.put` in and the `tasks_queue.put` action for the 4th part blocks the operation. 
So, we should not limit the size of the `tasks-queue` :
`self.tasks_queue = queue(num_threads)` => `self.tasks_queue = queue()`

Reply:
@ebozduman @vadmeste I agree with you, I did not consider that. As @ebozduman said we should remove the limit of queue size because we put all chunks of a file before we start the Worker threads.
thank you, guys.